massa-execution-worker/src/speculative_ledger.rs

```diff
@@ -123,7 +124,7 @@ impl SpeculativeLedger {
     ///
     /// # Returns
     /// `Some(Vec<u8>)` if the address was found, otherwise None
-    pub fn get_bytecode(&self, addr: &Address) -> Option<Vec<u8>> {
+    pub fn get_bytecode(&self, addr: &Address) -> Option<Bytecode> {
```

Comment:
I find a behavior a bit tricky there. We fetch the bytecode from ledger db and so it already have the version encoded inside the VecU8 and if we use his serializer it will re-add the version. It think the fact that we have, or not the version number of the bytecode should be uniform and comprehensible

Reply:
Right now the bytecode version is never included when we manipulate the `Vec<u8>` (needed in e.g. the interface in the execution module), so it should not be a problem until the `Bytecode`struct is trully versioned.
		
If I understand your comment correctly, I could preemptively implement the following methods to make the interface between `Vec<u8>` without version and `Bytecode` easier?
		
```rust
impl Bytecode {
	pub fn from_raw_bytes(version: u64, bytes: Vec<u8>) -> Self {}
	pub fn into_raw_bytes_without_version(&self) -> Vec<u8> {}			
}
```

Reply:
It's then if there is never the version in the `Vec<u8>` but is that a behavior that we want ? For example in `raw_get_bytecode_for` we will return a bytecode without version maybe in the future we will modify the shape of the bytecode by adding an extra byte and the user will not have any version byte to implement the new layout.

If you don't find an easy and uniform way now it's not a problem. The main goal of this PR is to able to keep the ledger for the next episodes and so changing the return format of an ABI isn't a problem.

Reply:
Ah I see what you mean, we might want to pass along the version byte everywhere (even the runtime, that only handles `Vec<u8>` and not `Bytecode`.
I'll look into it tomorrow, if I can't find a good way to handle that fast enough we will merge the current code then.

Reply:
👍